### ISSUE TITLE
too many controllers sent [sf#3413003]

### ISSUE THREAD

**Issue by Windfisch** [muse-sequencer/muse#274]
Submitted by spamatica on 2011-09-22 20:30:01

Am 20.09.2011 16:03, schrieb Florian Jung:

> Hi
> 
> i have a strange problem:
> i import a MIDI file which works well when played with another tool on
> another machine to the same synthesizer
> this MIDI (and also the import into muse2) doesn't contain any
> modulation controllers (only program and channel volume changes).
> 
> when i play it to my synthesizer, i'm always getting full modulation on
> all channels greater or equal than 3 (which sounds crappy). when i play
> to timidity as synth, it works fine. (and timidity DOES support
> modulation. if i set it manually, it modulates)
> 
> whenever i manually turn the modulation knob (it's always at zero. never
> moves!) to zero or one or something small, modulation is turned off. but
> once i re-start the song, or change the position manually, modulation is
> turned on again.
> even setting modulation to zero in my controller canvas does not always,
> bit sometimes help...
> 
> what's going on here?
> 
> is this related to my USB-Midi bridge? maybe data coming too fast or
> something like that? (shouldn't be, however. even if i set that channel
> solo, it has modulation)
> 
> can anyone please tell my how to fix it or how to investigate it further?
> 
> seems to be a bug in muse: when muse uses my synthe's midi port directly
> (via alsa), it shows these problems. if i however use a2jmidid (an jack
> to alsa midi bridge), and use that bridge's jack ports from muse, it works!

i thought, this could possibly be related to the MIDI data rate. maybe
muse simply sends events without watching out for the maximum data rate
MIDI can carry and so sends garbage. a2jmidid maybe watches out and
works for that reason.
however, i removed all tracks except one, and it still shows that bug
when playing via muse's midi port directly.

it also seems not to be related to alsa midi ports in general, as
timidity also uses an alsa midi port and works well

to sum it up: i have absolutely no idea what's the bug or why a2jmidid
fixes it. but i think we should find the bug and squash it!
